### PR TITLE
[AMDGPU] Add test to show s_cselect generation from uniform select

### PR DESCRIPTION
**Pull request by choikwa**

### CODE REVIEW

**Comment by choikwa** [comment]
@bcahoon @jrbyrnes Could you review this change please? Thank you

**Comment by jrbyrnes** [comment]
Can you fix the commit title split, and also prepend the commit title with [AMDGPU]?

**Comment by choikwa** [comment]
Addressed comments and rebased with 816f14d94166

File: llvm/test/CodeGen/AMDGPU/insert_extract_element.ll [hunk 1]
```diff
@@ -0,0 +1,218 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
+; RUN: llc -mtriple=amdgcn--amdhsa -mcpu=gfx90a -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX90a %s
```
Comment (jayfoad) [inline llvm/test/CodeGen/AMDGPU/insert_extract_element.ll:2]: Nit: capital A in GFX90A please

**Review by jayfoad — COMMENTED** [review]
The test looks OK, but what is it testing, and what does it have to do with V_CNDMASK_B32?

File: llvm/test/CodeGen/AMDGPU/insert_extract_element.ll (hunk 1 above)
Comment (arsenm) [reply llvm/test/CodeGen/AMDGPU/insert_extract_element.ll:2]: Don't need -verify-machineinstrs

File: llvm/test/CodeGen/AMDGPU/insert_extract_element.ll [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+  br label %for.body.i.i
+
+for.body.i.i:                                     ; preds = %for.body.i.i, %entry
+  %x.sroa.0.036.i.i = phi <4 x i32> [ %0, %entry ], [ %4, %for.body.i.i ]
+  %X.sroa.0.035.i.i = phi <4 x i32> [ zeroinitializer, %entry ], [ %2, %for.body.i.i ]
+  %idxprom.i.i = zext i32 %dr.037.i.i to i64
+  %1 = extractelement <4 x i32> %X.sroa.0.035.i.i, i64 %idxprom.i.i
+  %add.i.i = or i32 %1, %inc.i.i
+  %2 = insertelement <4 x i32> %X.sroa.0.035.i.i, i32 %add.i.i, i64 %idxprom.i.i
+  %3 = extractelement <4 x i32> %x.sroa.0.036.i.i, i64 %idxprom.i.i
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/insert_extract_element.ll:215]: Use named values in test

File: llvm/test/CodeGen/AMDGPU/insert_extract_element.ll [hunk 3]
```diff
@@ -0,0 +1,218 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
+; RUN: llc -mtriple=amdgcn--amdhsa -mcpu=gfx90a -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX90a %s
+; RUN: llc -mtriple=amdgcn--amdhsa -mcpu=gfx940 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX940 %s
+; RUN: llc -mtriple=amdgcn--amdhsa -mcpu=gfx1030 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX1030 %s
+; RUN: llc -mtriple=amdgcn--amdhsa -mcpu=gfx1100 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX1100 %s
+
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/insert_extract_element.ll:6]: Test name is very general, and we already have tests for insertelement/extractelement. What is this testing that's new? Can it be merged into an existing test?

**Comment by choikwa** [comment]
> The test looks OK, but what is it testing, and what does it have to do with V_CNDMASK_B32?

The test was reduced from a QUDA application with which rocm 5.5.1 was generating V_CNDMASK with an illegal operand type. This was found to be from a partial commit in attempt to reland another change. LLVM trunk no longer generates the instruction but I still thought it would be a good test to have.

File: llvm/test/CodeGen/AMDGPU/insert_extract_element.ll (hunk 3 above)
Comment (choikwa) [reply llvm/test/CodeGen/AMDGPU/insert_extract_element.ll:6]: insert_vector_elt.ll looks like a good candidate. Will include in there instead.

File: llvm/test/CodeGen/AMDGPU/insert_extract_element.ll (hunk 1 above)
Comment (arsenm) [reply llvm/test/CodeGen/AMDGPU/insert_extract_element.ll]: No, march names are canonically all lowercase

Comment (arsenm) [reply llvm/test/CodeGen/AMDGPU/insert_extract_element.ll]: Oh, for the check prefix. That should be all caps

File: llvm/test/CodeGen/AMDGPU/insert_extract_element.ll (hunk 3 above)
Comment (arsenm) [reply llvm/test/CodeGen/AMDGPU/insert_extract_element.ll]: It's still not really a test of this. If it's going to be standalone should name it something relevant to the problem you were looking for

**Comment by arsenm** [comment]
> > The test looks OK, but what is it testing, and what does it have to do with V_CNDMASK_B32?
> 
> The test was reduced from a QUDA application with which rocm 5.5.1 was generating V_CNDMASK with an illegal operand type. This was found to be from a partial commit in attempt to reland another change. LLVM trunk no longer generates the instruction but I still thought it would be a good test to have.

Do you know what commit fixed it?

**Comment by choikwa** [comment]
fixed by fbdea5a2e9169

**Comment by choikwa** [comment]
Changed title and testcase to uniform-select.ll per @jrbyrnes' suggestion

**Comment by choikwa** [comment]
latest update renames numbered variables

**Review by jrbyrnes — APPROVED** [review]
LGTM

**Pull request merged**